### DISCUSSION TITLE
Convert DATADISK to ROOT

### DISCUSSION THREAD

**Discussion by top-secrett** [apache/cloudstack#9710] [Q&A] [answered]
##### ISSUE TYPE

 * Improvement Request
 * Enhancement Request
 * Feature Idea

For what reason CloudStack doesn't allow to convert DATADISK to ROOT? It is very usefull for backup purpose.
I would be great to create volume snapshot via Recurring Snapshots, after that create volume and attach it to Intsance instead of current ROOT volume

**Comment by weizhouapache** [comment, marked as answer]
@top-secrett 
When you attacha DATADISK, you can specify the deviceid 
if deviceid is 0, the DATADISK will be attached as a ROOT disk. It works only if the vm do not have a ROOT disk, as a vm can have only 1 ROOT disk.

**Comment by top-secrett** [comment]
@weizhouapache oh, really. Earlier I got error. Thank you
